src/components/media-controls/media-controls.ts

```diff
@@ -181,225 +180,40 @@ export class MediaControlsComponent extends AbstractComponent(LitElement) {
-    return html`
-      <sl-menu-item>
-        ${text}
-        <sl-menu slot="submenu">
-          <label>
-            <input
-              @change="${changeHandler}"
-              type="range"
-              min="${min}"
-              max="${max}"
-              step="${step}"
-              value="${currentValue}"
-            />
-          </label>
-        </sl-menu>
-      </sl-menu-item>
-    `;
-  }
-
   private settingsTemplate() {
-    if (!this.spectrogramElement) {
+    if (!this.spectrogramElement || !this.axesElement) {
       return nothing;
     }
 
-    const possibleWindowSizes = this.spectrogramElement.possibleWindowSizes;
-    const possibleWindowOverlaps = this.spectrogramElement.possibleWindowOverlaps;
-    const currentOptions = this.spectrogramElement.spectrogramOptions;
-
-    const discreteDropdownHandler = (key: keyof SpectrogramOptions) => {
-      return (event: CustomEvent<{ item: SlMenuItem }>) => {
-        if (!this.spectrogramElement) {
-          throw new Error("No spectrogram element found");
-        }
-
-        // TODO: remove this after demo
-        let newValue: string | number | boolean = ["windowSize", "windowOverlap"].includes(key)
-          ? Number(event.detail.item.value)
-          : event.detail.item.value;
-
-        if (key === "melScale") {
-          newValue = newValue === "mel";
-        }
-
-        const oldOptions = this.spectrogramElement.spectrogramOptions;
-        if (key === "windowSize" && this.spectrogramElement) {
-          if (this.spectrogramElement.spectrogramOptions.windowOverlap >= (newValue as number)) {
-            oldOptions.windowOverlap = (newValue as number) / 2;
-          }
-        }
-
-        this.spectrogramElement.spectrogramOptions = {
-          ...oldOptions,
-          [key]: newValue,
-        } as any;
-
-        this.requestUpdate();
-      };
-    };
-
-    const rangeInputHandler = (key: keyof SpectrogramOptions) => {
-      return (event: Event) => {
-        if (!this.spectrogramElement) {
-          throw new Error("No spectrogram element found");
-        }
-
-        const newValue = (event.target as HTMLInputElement).value;
-        const oldOptions = this.spectrogramElement.spectrogramOptions;
-
-        this.spectrogramElement.spectrogramOptions = {
-          ...oldOptions,
-          [key]: Number(newValue),
-        } as any;
-      };
-    };
-
-    const axesChangeHandler = (event: CustomEvent<{ item: SlMenuItem }>) => {
-      if (!this.axesElement) {
-        throw new Error("No axes element found");
+    const spectrogramOptionsSignal = signal<SpectrogramOptions>(this.spectrogramElement.spectrogramOptions);
+    spectrogramOptionsSignal.subscribe((newOptions) => {
+      if (!this.spectrogramElement) {
+        throw new Error(
+          "No spectrogram element found. This might be because you forgot to unsubscribe from the options signal",
+        );
       }
+      this.spectrogramElement.spectrogramOptions = newOptions;
```

Comment:
this seems to be a recursive definition.

but also, why is it the job of media controls to set the options on a spectrogram?